src/lib/routers/history.ts

```diff
@@ -0,0 +1,200 @@
+import qs from 'qs';
+import { RouteState } from '../../types';
+
+type BrowserHistoryProps = {
+  windowTitle?: (routeState: RouteState) => string;
+  writeDelay: number;
+  createURL({
+    qsModule,
+    routeState,
+    location,
+  }: {
+    qsModule: typeof qs;
+    routeState: RouteState;
+    location: Location;
+  }): string;
+  parseURL({
+    qsModule,
+    location,
+  }: {
+    qsModule: typeof qs;
+    location: Location;
+  }): RouteState;
+};
+
+const defaultCreateURL: BrowserHistoryProps['createURL'] = ({
```

Comment:
IIRC we lose the inference with this notation sometimes. We could rather inline the type directly. It won't compile if there is a mismatch anyway. This is especially true if we extract the value of the function into its own type.